apps/cowswap-frontend/src/legacy/state/orders/middleware/showPendingOrderNotification.tsx

```diff
@@ -0,0 +1,44 @@
+import { jotaiStore } from '@cowprotocol/core'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { addSnackbarAtom } from '@cowprotocol/snackbars'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import { Nullish } from 'types'
+
+import { PendingOrderNotification } from 'common/containers/PendingOrderNotification'
+import { UiOrderType } from 'utils/orderUtils/getUiOrderType'
+
+interface PendingOrderNotificationParams {
+  id: string
+  owner: string
+  kind: OrderKind
+  uiOrderType: UiOrderType
+  receiver: Nullish<string>
+  inputAmount: CurrencyAmount<Token>
+  outputAmount: CurrencyAmount<Token>
+  orderCreationHash?: string
+  isHidden?: boolean
+}
+
+export function showPendingOrderNotification(order: PendingOrderNotificationParams) {
+  const { id, receiver, owner, uiOrderType, orderCreationHash, isHidden, inputAmount, outputAmount } = order
+
+  if (!isHidden) {
+    const hasCustomRecipient = receiver?.toLowerCase() !== owner.toLowerCase()
+
+    jotaiStore.set(addSnackbarAtom, {
```

Comment:
So, mainly I cut a corner here.
Maybe it would be better to use `useAddSnackbar` instead, but it requires too much refactoring.